CHANGES.rst

```diff
@@ -2,6 +2,14 @@
 Changes
 =======
 
+0.4.0
+-----
+ 
+ * Breaking change: `expandvars_dict` drops support for multilevel variable expansion introduced in 0.2.
+   Use multiple variables in a single definition instead. e.g. `http://${HOST}:${PORT}`.
```

Comment:
I don't think the use case for multilevel variables was "I want to have different variables in a single variable", but AFAIR it was something along the lines of "app.json for review apps has something, that is slightly different than what we have in production, and it has to be expanded twice". I'll need to dig through history to see what the exact use case was and if we still need it.

Reply:
Found it in https://github.com/niteoweb/dmon/issues/367#issuecomment-292933614:

<img width="785" alt="Screenshot 2019-05-23 at 12 54 35" src="https://user-images.githubusercontent.com/311580/58247421-f0619880-7d59-11e9-99f2-f9cc9c471865.png">

The idea certainly is that EBN starts using pyramid_heroku, so that we remove similar code from EBN's codebase. But that means pyramid_heroku needs to support all EBN's use cases.

Reply:
And yes, EBN has a similar env var in Heroku: `CDN_BASE_URL:  https://cdn.easyblognetworks.com/${HEROKU_RELEASE_VERSION}`

Reply:
This is done in this way so that CDN files can be cached forever, since on every new Heroku deploy/release, the CDN URL changes.

Reply:
Why can't we define like this?
```
fanstatic.base_url =  https://cdn.easyblognetworks.com/${HEROKU_RELEASE_VERSION}
```
Instead of
```
fanstatic.base_url = ${CDN_BASE_URL}
```

Reply:
Right with default expandvars it's not possible.

Reply:
This is the breaking change I mentioned.

Reply:
Okay have to remove the word `multiple variables` and use `variables` instead.

Reply:
Updated the wording.

Reply:
> Why can't we define like this?
> `fanstatic.base_url =  https://cdn.easyblognetworks.com/${HEROKU_RELEASE_VERSION}`

I don't know? Seems like a good idea to try out! Might be a 🤦‍♂on my side, too obvious for me to see a simple solution.

Reply:
> Updated the wording.

This doesn't help with missing functionality :)

Reply:
Actually I wrote this alternate library specifically for ebn. :smile:  In ebn, we only need to change one env variable. Then it should work right?

Reply:
To make the change safely, we can introduce a new env variable, make the migration, then remove the old variables.

Reply:
Or a file named `.heroku/reviewapp.sh` that does:

```bash
#!/usr/bin/env bash

echo CDN_BASE_URL=https//${HEROKU_APP_NAME}.herokuapp.com/ >> /etc/environment
# Or
echo CDN_BASE_URL=https//${HEROKU_APP_NAME}.herokuapp.com/ >> ~/.bashrc

./.heroku/reviewappdb.sh
```

Reply:
Okay `echo CDN_BASE_URL=https//${HEROKU_APP_NAME}.herokuapp.com/ >> ~/.bashrc` may not work
but `echo CDN_BASE_URL=https//${HEROKU_APP_NAME}.herokuapp.com/ >> /etc/environment` should work.

Reply:
Ok it doesn't work.

Reply:
Finally [another approach worked](https://github.com/niteoweb/ebn/pull/6033/files) 😎